base_layer/core/src/transactions/coinbase_builder.rs

```diff
@@ -255,5 +255,82 @@ mod test {
             .open_value(&p.spend_key, block_reward.into(), utxo.commitment()));
         assert!(utxo.verify_range_proof(&factories.range_proof).unwrap());
         assert!(utxo.features.flags.contains(OutputFlags::COINBASE_OUTPUT));
+        assert_eq!(
+            tx.body.check_coinbase_output(
+                block_reward,
+                rules.consensus_constants().coinbase_lock_height(),
+                &factories,
+                42
+            ),
+            Ok(())
+        );
+    }
+
+    #[test]
+    fn invalid_coinbase() {
+        let p = TestParams::new();
+        let (builder, rules, factories) = get_builder();
+        // We just want some small amount here.
+        let missing_fee = rules.emission_schedule().block_reward(4200000) + (2 * uT);
+        let builder = builder
+            .with_block_height(42)
+            .with_fees(1 * uT)
+            .with_nonce(p.nonce.clone())
+            .with_spend_key(p.spend_key.clone());
+        let (mut tx, _) = builder
+            .build(rules.consensus_constants(), rules.emission_schedule())
+            .unwrap();
+        let block_reward = rules.emission_schedule().block_reward(42) + missing_fee;
+        let builder = CoinbaseBuilder::new(factories.clone());
+        let builder = builder
+            .with_block_height(4200000)
+            .with_fees(1 * uT)
+            .with_nonce(p.nonce.clone())
+            .with_spend_key(p.spend_key.clone());
+        let (tx2, _) = builder
+            .build(rules.consensus_constants(), rules.emission_schedule())
+            .unwrap();
+        let mut coinbase2 = tx2.body.outputs()[0].clone();
+        let mut coinbase_kernel2 = tx2.body.kernels()[0].clone();
+        coinbase2.features = OutputFeatures::default();
+        coinbase_kernel2.features = KernelFeatures::empty();
+        tx.body.add_output(coinbase2);
+        tx.body.add_kernel(coinbase_kernel2);
+
+        // test catches that coinbase amount is wrong
+        assert_eq!(
+            tx.body.check_coinbase_output(
+                block_reward,
+                rules.consensus_constants().coinbase_lock_height(),
+                &factories,
+                42
+            ),
+            Err(TransactionError::InvalidCoinbase)
+        );
+        // testing that "block" is still valid
+        assert_eq!(
+            tx.body
+                .validate_internal_consistency(&BlindingFactor::default(), block_reward, &factories),
+            Ok(())
+        );
```

Comment:
For sanity, can we add a call to one of the validators that catches the inflation created here?